File: year2020/month11/day1127/ms_1127.cs
using System;

namespace algorithm_110
{
    class ms_1127
    {
        static int[] SizeUp(int[] data)
        {
            int len = data.Length;
            int[] temp = new int[len + 1];
            int i = 0;

            for(i = 0; i < len; i++)
            {
                temp[i] = data[i];
            }

            temp[i] = -1;

            return temp;
        }

        static void Main(string[] args)
        {
            String[] input = Console.ReadLine().Split(' ');
            int len = input.Length;
            int[] numbers = new int[len];

            if(2 <= len && len <= 100)
            {
                for(int i = 0; i < input.Length; i++)
                {
                    int temp = Convert.ToInt32(input[i]);

                    if(0 <= temp && temp <= 100)
                    {
                        numbers[i] = temp;
                    }
                }
            }

            int[] answer = new int[1] { -1 };
            int count = 0;

            for(int i = 0; i < numbers.Length - 1; i++)
            {
                for(int j = i+1; j < numbers.Length; j++)
                {
                    int insertValue = numbers[i] + numbers[j];
                    bool isIn = false;

                    if(answer.Length == count)
                    {
                        answer = SizeUp(answer);
                    }

                    for(int h = 0; h < answer.Length; h++)
                    {
                        if(answer[h] == insertValue)
                        {
                            isIn = true;
                        }
                    }

                    if(!isIn)
                    {
                        answer[count] = insertValue;
                        count++;
                    }
                }
            }
            
            Array.Sort(answer);
            
            for(int i = 0; i < answer.Length; i++)
            {
                if(answer[i] != -1) // 가끔 사이즈를 늘려서 -1이 들어가있을 떄를 위한 작업 ㅠㅠ
                {
                    Console.Write(answer[i]);
            
                    if(i != count)
                    {
                        Console.Write(' ');
                    }
                }
                else continue;
            }
        }
    }
}

File: year2020/month11/day1127/ms_1127.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;

int main() {
	string input;
	vector<int> numbers;

	getline(cin, input);

	stringstream ss(input);
	int len = 0;

	while (getline(ss, input, ' ')) {
		int temp = stoi(input);

		if (0 <= temp && temp <= 100) {
			numbers.push_back(temp);
			len++;
		}
	}

	vector<int> answer;
	int count = 0;

	// insert sum of two value to answer
	if (2 <= len && len <= 100) {
		for (int i = 0; i < len - 1; i++) {
			for (int j = i + 1; j < len; j++) {
				int temp = numbers[i] + numbers[j];
				bool isIn = false;

				for (int h = 0; h < answer.size(); h++) {
					if (answer[h] == temp) {
						isIn = true;
					}
				}
				
				if (!isIn) {
					answer.push_back(temp);
					count++;
				}
			}
		}

		// selection sort
		for (int i = 0; i < count - 1; i++) {
			int index = i;

			for (int j = i + 1; j < count; j++) {
				if (answer[index] > answer[j]) {
					index = j;
				}
			}

			int temp = answer[index];
			answer[index] = answer[i];
			answer[i] = temp;
		}

		// print answer
		for (int i = 0; i < count; i++) {
			cout << answer[i];

			if (i != count) {
				cout << ' ';
			}
		}
	}

	return 0;
}